File: feature/base/src/main/java/net/twisterrob/android/view/color/swatches/pixel/color/RadialHueGradient.java
package net.twisterrob.android.view.color.swatches.pixel.color;

import net.twisterrob.android.view.color.*;

public class RadialHueGradient implements PixelColor {
	private int w, h;

	@Override
	public void initializeInvariants(int w, int h) {
		this.w = w;
		this.h = h;
	}

	@Override
	public int getPixelColorAt(int x, int y) {
		float angle = FastMath.Atan2Faster.atan2(y - w / 2, x - h / 2) + ColorMath.PI;
		float hue = angle / (ColorMath.PI * 2);
		return ColorMath.fromHsb(hue, 1, 1, 1);
	}
}

File: feature/base/src/main/java/net/twisterrob/android/view/color/swatches/pixel/color/RadialHueGradient.kt
package net.twisterrob.android.view.color.swatches.pixel.color

import net.twisterrob.android.view.color.FastMath
import net.twisterrob.android.view.color.PI
import net.twisterrob.android.view.color.fromHsb

class RadialHueGradient : PixelColor {

	private var w: Int = 0
	private var h: Int = 0

	override fun initializeInvariants(w: Int, h: Int) {
		this.w = w
		this.h = h
	}

	override fun getPixelColorAt(x: Int, y: Int): Int {
		val angle = FastMath.Atan2Faster.atan2((y - w / 2).toFloat(), (x - h / 2).toFloat()) + PI
		val hue = angle / (PI * 2)
		return fromHsb(hue, 1f, 1f, 1f)
	}
}
